### PR TITLE
Misra 11.x: pointer usage.

### PR DESCRIPTION
**Pull request by rbiasini**
Also replace CAN byte manipulation with GET-BYTE macro

### CODE REVIEW

File: board/drivers/llcan.h [hunk 1]
```diff
@@ -8,6 +8,13 @@
 // 5000 = 500 kbps
 #define can_speed_to_prescaler(x) (CAN_PCLK / CAN_QUANTA * 10U / (x))
 
+#define GET_BUS(msg) (((msg)->RDTR >> 4) & 0xFF)
+#define GET_LEN(msg) ((msg)->RDTR & 0xf)
```
Comment (quillford) [inline board/drivers/llcan.h]: `0xF` for consistency

File: board/safety/safety_cadillac.h [hunk 2]
```diff
@@ -28,20 +28,21 @@ static void cadillac_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
...
+
     torque_driver_new = to_signed(torque_driver_new, 11);
     // update array of samples
     update_sample(&cadillac_torque_driver, torque_driver_new);
   }
 
   // this message isn't all zeros when ignition is on
   if ((addr == 0x160) && (bus == 0)) {
-    cadillac_ign = to_push->RDLR > 0;
+    cadillac_ign = GET_BYTE(to_push, 0) | GET_BYTE(to_push, 1) | GET_BYTE(to_push, 2) | GET_BYTE(to_push, 3);
```
Comment (quillford) [inline board/safety/safety_cadillac.h]: why not use `GET_BYTES_04`?

File: board/safety/safety_honda.h [hunk 3]
```diff
@@ -22,13 +22,13 @@ static void honda_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
   // sample speed
   if (addr == 0x158) {
     // first 2 bytes
-    honda_ego_speed = to_push->RDLR & 0xFFFF;
+    honda_ego_speed = GET_BYTE(to_push, 0) | GET_BYTE(to_push, 1);
```
Comment (quillford) [inline board/safety/safety_honda.h]: maybe `honda_moving` is a more appropriate name?

Comment (rbiasini) [reply board/safety/safety_honda.h]: done

File: board/drivers/llcan.h (hunk 1 above)
Comment (rbiasini) [reply board/drivers/llcan.h]: done

File: board/safety/safety_cadillac.h (hunk 2 above)
Comment (rbiasini) [reply board/safety/safety_cadillac.h]: done

File: board/provision.h [hunk 4]
```diff
@@ -5,7 +5,7 @@
 // SHA1 checksum = 0x1C - 0x20
 
 void get_provision_chunk(uint8_t *resp) {
-  (void)memcpy(resp, (void *)0x1fff79e0, PROVISION_CHUNK_LEN);
+  (void)memcpy(resp, (uint8_t *)0x1fff79e0, PROVISION_CHUNK_LEN);
```
Comment (rbiasini) [inline board/provision.h]: this was failing 11.6: A cast shall not be performed between pointer to void and an arithmetic type

File: board/safety/safety_cadillac.h [hunk 5]
```diff
@@ -28,20 +28,21 @@ static void cadillac_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
...
+
     torque_driver_new = to_signed(torque_driver_new, 11);
     // update array of samples
     update_sample(&cadillac_torque_driver, torque_driver_new);
   }
 
   // this message isn't all zeros when ignition is on
   if ((addr == 0x160) && (bus == 0)) {
-    cadillac_ign = to_push->RDLR > 0;
+    cadillac_ign = GET_BYTES_04(to_push);
```
Comment (geohot) [inline board/safety/safety_cadillac.h]: != 0

File: board/safety/safety_ford.h [hunk 6]
```diff
@@ -9,22 +9,25 @@
...
 
   int addr = GET_ADDR(to_push);
 
   if (addr == 0x217) {
     // wheel speeds are 14 bits every 16
-    ford_is_moving = 0xFCFF & (to_push->RDLR | (to_push->RDLR >> 16) |
-                               to_push->RDHR | (to_push->RDHR >> 16));
+    ford_moving = false;
+    for (int i = 0; i < 8; i += 2) {
+      int j = i + 1;  // Misra 10.8: can't cast (i+1) directly in uint
```
Comment (geohot) [inline board/safety/safety_ford.h]: Don't do this!

File: board/spi_flasher.h [hunk 7]
```diff
@@ -181,9 +181,9 @@ void bl_can_send(uint8_t *odat) {
 void CAN1_RX0_IRQHandler(void) {
   while (CAN->RF0R & CAN_RF0R_FMP0) {
     if ((CAN->sFIFOMailBox[0].RIR>>21) == CAN_BL_INPUT) {
-      uint8_t dat[8];
-      ((uint32_t*)dat)[0] = CAN->sFIFOMailBox[0].RDLR;
-      ((uint32_t*)dat)[1] = CAN->sFIFOMailBox[0].RDHR;
+      uint32_t dat[2];
```
Comment (geohot) [inline board/spi_flasher.h]: BUG! See line 205

File: board/safety/safety_honda.h [hunk 8]
```diff
@@ -49,13 +49,13 @@ static void honda_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
   // most hondas: 0x17C bit 53
   // accord, crv: 0x1BE bit 4
   #define IS_USER_BRAKE_MSG(addr) (!honda_alt_brake_msg ? ((addr) == 0x17C) : ((addr) == 0x1BE))
```
Comment (geohot) [inline board/safety/safety_honda.h]: Don't make this macro for one time use!

File: board/safety/safety_gm_ascm.h [hunk 9]
```diff
@@ -13,7 +13,7 @@ static int gm_ascm_fwd_hook(int bus_num, CAN_FIFOMailBox_TypeDef *to_fwd) {
     // block 0x315 and 0x2cb, which are the brake and accel commands from ASCM1
     //if ((addr == 0x152) || (addr == 0x154) || (addr == 0x315) || (addr == 0x2cb)) {
     if ((addr == 0x152) || (addr == 0x154)) {
-      int supercruise_on = (to_fwd->RDHR >> 4) & 0x1;  // bit 36
+      int supercruise_on = GET_BYTE(to_fwd, 4) & 0x10;  // bit 36
```
Comment (geohot) [inline board/safety/safety_gm_ascm.h]: bool? != 0

File: board/drivers/usb.h [hunk 10]
```diff
@@ -404,7 +404,7 @@ void *USB_ReadPacket(void *dest, uint16_t len) {
   return ((void *)dest_copy);
 }
 
-void USB_WritePacket(const uint8_t *src, uint16_t len, uint32_t ep) {
+void USB_WritePacket(void *src, uint16_t len, uint32_t ep) {
```
Comment (geohot) [inline board/drivers/usb.h]: const void *src, and fix above

File: board/safety/safety_gm_ascm.h (hunk 9 above)
Comment (rbiasini) [reply board/safety/safety_gm_ascm.h]: ok. There are a bunch of these implicit int->bool casts. Will fix the rest in a different PR.

File: board/drivers/usb.h (hunk 10 above)
Comment (rbiasini) [reply board/drivers/usb.h]: fixed

File: board/safety/safety_honda.h (hunk 8 above)
Comment (rbiasini) [reply board/safety/safety_honda.h]: done

File: board/spi_flasher.h (hunk 7 above)
Comment (rbiasini) [reply board/spi_flasher.h]: gh, misundersood what `((uint32_t*)dat)[0] = CAN->sFIFOMailBox[0].RDLR;` does. Should be ok now.

File: board/safety/safety_ford.h (hunk 6 above)
Comment (rbiasini) [reply board/safety/safety_ford.h]: solution is still a bit dumb, but won't trigger Misra

File: board/safety/safety_cadillac.h (hunk 5 above)
Comment (rbiasini) [reply board/safety/safety_cadillac.h]: Done

**Pull request merged**